extensions/wikia/Email/MobileApplicationsLinksGenerator.class.php

```diff
@@ -0,0 +1,174 @@
+
+	const APPLICATION_APP_STORE_URL = 'https://itunes.apple.com/%s/app/id%s';
+
+	const WIKIA_GOOGLE_PALY_URL = 'https://play.google.com/store/apps/developer?id=Wikia,+Inc.&hl=%s';
+
+	const APPLICATION_GOOGLE_PLAY_URL = 'https://play.google.com/store/apps/details?hl=%s&id=%s';
+
+	const ANDROID_PLATFORM = 'android';
+
+	const IOS_PLATFORM = 'ios';
+
+	const MOBILE_APPLICATIONS_LINKS_EVICTION_TIME = 24 * 60 * 60; // 24h
+
+	static private $appStoreLanguagesMapping = [
+		'en' => 'us',
+		'zh' => 'cn',
+		'zh-tw' => 'tw',
+		'fr' => 'fr',
+		'de' => 'de',
+		'it' => 'it',
+		'ja' => 'jp',
+		'pl' => 'pl',
+		'pt' => 'pt',
+		'pt-br' => 'br',
+		'ru' => 'ru',
+		'es' => 'es',
+	];
+
+	/**
+	 * Generates links for mobile applications for current wiki.
+	 *
+	 * @param string $storeLanguage - language for localized store
+	 *
+	 * @return array - currently will contains 2 elements, link for app store and google play
+	 */
+	public function generate( $storeLanguage ) {
+		global $wgMemc;
+
+		$mobileApplicationsLinks =
+			$wgMemc->get( $this->createCacheKeyForMobileApplicationsLinks() );
+
+		if ( is_bool( $mobileApplicationsLinks ) ) {
+			$mobileApplicationsLinks = [];
+			$response = $this->fetchMobileApplicationsDetails();
+
+			if ( $response && $this->applicationsExistIn( $response ) ) {
+				$mobileApplications = json_decode( $response, true );
+				$mobileApplicationsLinks =
+					$this->traverseThrough( $mobileApplications, $storeLanguage );
+			}
+
+			$mobileApplicationsLinks =
+				$this->fillWithWikiaAccountLinksIfEmpty( $mobileApplicationsLinks, $storeLanguage );
+
+			// Even if response is not valid (for example there are some difficulties with mobile applications service)
+			// result will be cached to prevent too long delays in email generation
+			$wgMemc->set( $this->createCacheKeyForMobileApplicationsLinks(),
+				$mobileApplicationsLinks, time() + self::MOBILE_APPLICATIONS_LINKS_EVICTION_TIME );
+		}
+
+		return $mobileApplicationsLinks;
+	}
+
+	/**
+	 * Fills array with wikia mobile application store link if application was not found
+	 *
+	 * @param array $mobileApplicationsLinks
+	 * @param string $storeLanguage
+	 * @return array with links to app store and google play
+	 */
+	private function fillWithWikiaAccountLinksIfEmpty( $mobileApplicationsLinks, $storeLanguage ) {
+		if ( empty ( $mobileApplicationsLinks[self::ANDROID_PLATFORM] ) ) {
+			$mobileApplicationsLinks[self::ANDROID_PLATFORM] =
+				sprintf( self::WIKIA_GOOGLE_PALY_URL,
+					$this->prepareLanguageForGooglePlay( $storeLanguage ) );
+		}
+
+		if ( empty( $mobileApplicationsLinks[self::IOS_PLATFORM] ) ) {
+			$mobileApplicationsLinks[self::IOS_PLATFORM] =
+				sprintf( self::WIKIA_APP_STORE_URL,
+					$this->prepareLanguageForAppStore( $storeLanguage ) );
+		}
+
+		return $mobileApplicationsLinks;
+	}
+
+	/**
+	 * @param string $language
+	 * @return string
+	 */
+	private function prepareLanguageForGooglePlay( $language ) {
```

Comment:
I understand that this method is meant to mirror the `prepareLanguageForAppStore` method, but I'm not sure that  symmetry is a great reason for an identity method to exist.  What if you brought the whole `sprintf` info both methods, and renamed them`generateGooglePlayURL` and `generateAppStoreURL`?